src/peft/peft_model.py

```diff
@@ -505,7 +505,14 @@ def get_nb_trainable_parameters(self) -> tuple[int, int]:
             # one needs to multiply the number of parameters by 2 to get
             # the correct number of parameters
             if param.__class__.__name__ == "Params4bit":
-                num_params = num_params * 2
+                dtype2bytes = {}
+                dtype2bytes[torch.float32] = 4
+                dtype2bytes[torch.float16] = 2
+                dtype2bytes[torch.bfloat16] = 2
+                dtype2bytes[torch.uint8] = 1
+                dtype2bytes[torch.int8] = 1
```

Comment:
Instead of using this dict, can we not use `param.quant_storage.itemsize`? Also, `quant_storage` seems to be a new attribute. Should we hard-code the bytes to 2 if the attribute does not exist or fix the min version of bnb?